inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/InventoryFileWatcher.scala

```diff
@@ -0,0 +1,167 @@
+/*
+*************************************************************************************
+* Copyright 2019 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+
+package com.normation.inventory.provisioning.endpoint
+
+import better.files._
+import net.liftweb.common.Full
+import monix.execution.Scheduler._
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.control.NonFatal
+
+/**
+ * That class is in charge of watching file creation on a repos and process
+ * inventory file as soon as an inventory+signature file is created.
+ */
+class InventoryFileWatcher(
+    inventoryProcessor   : InventoryProcessor
+  , incomingInventoryPath: String
```

Comment:
There is two incoming inventory directories: 
- /var/rudder/inventories/incoming => new node with initial promises
- /var/rudder/inventories/accepted-nodes-updates => node with at least one promise update. 

We need to take care of both, of course.